src/compiler/compile/render_dom/wrappers/AwaitBlock.ts

```diff
@@ -188,6 +188,7 @@ export default class AwaitBlockWrapper extends Wrapper {
 			ctx: #ctx,
 			current: null,
 			token: null,
+			hasCatch: false,
```

Comment:
```suggestion
			hasCatch: ${this.catch.node.start !== 'null' ? 'true' : 'false'},
```

Reply:
since we can determine `hasCatch` on build time

Reply:
Would this then also involve removing the code added below at line 204?

Reply:
yes.

Reply:
Looks better, Let me change it.

Reply:
Done. Please take a look again @tanhauhau